### ISSUE TITLE
Time-series models

### ISSUE THREAD

**Issue by valer1435** [valer1435/RepoPilot#16]
Which models for time series forecasting fedot uses? Can you provide examples? Is there any comparison with competitors? @RepoPilotAssistant please help

**Comment by RepoPilotAssistant** [comment]
To compare Fedot with other AutoML and time-series forecasting tools available in the market, you can consider tools like Auto-sklearn, H2O.ai, and TPOT. Each of these tools has its own strengths and weaknesses. Auto-sklearn is known for its integration with the scikit-learn ecosystem, H2O.ai offers a comprehensive platform with a wide range of models and algorithms, and TPOT focuses on genetic programming for pipeline optimization. Comparing these tools with Fedot would involve evaluating their model variety, performance, and ease of use, as well as their specific capabilities for time-series forecasting.

**Comment by valer1435** [comment]
At which benchmarks fedot was tested? @RepoPilotAssistant

**Comment by RepoPilotAssistant** [comment]
Fedot was tested at benchmarks using the Symmetric Mean Absolute Percentage Error (SMAPE) metric on a subsample from the M4 competition, which includes 998 series with various intervals (daily, weekly, monthly, quarterly, yearly). The forecasting horizons vary by series type: 6 for yearly, 8 for quarterly, 18 for monthly, 13 for weekly, and 14 for daily. Fedot's performance was compared with other libraries such as AutoTS, TPOT, H2O, pmdarima, and Autogluon. The results show that Fedot's SMAPE values are competitive, with the best values highlighted in bold for each interval. For instance, at the 50th quantile, Fedot's SMAPE values are 2.326 for daily, 4.95 for weekly, 7.123 for monthly, 6.786 for quarterly, and 8.682 for yearly series. Statistical analysis using the Friedman t-test confirms that Fedot's time series forecasting ability is statistically indistinguishable from the field leaders, represented by Autogluon and NBEATS.

**Comment by valer1435** [comment]
Can you give an example how to forecast time series with custom pipeline? @RepoPilotAssistant please help

**Comment by valer1435** [comment]
Can you give an example how to forecast time series with custom pipeline? @RepoPilotAssistant please help

**Comment by valer1435** [comment]
Can you give an example how to forecast time series with custom pipeline? @RepoPilotAssistant please help

**Comment by valer1435** [comment]
How to forecast time series with custom pipeline? @RepoPilotAssistant

**Comment by valer1435** [comment]
How to forecast time series with custom pipeline? @RepoPilotAssistant

**Comment by RepoPilotAssistant** [comment]
To forecast time series with a custom pipeline in Fedot, you can manually compose your own pipeline using the `PipelineBuilder`. Here's an example of how to do this:

1. **Import necessary modules**:
   ```python
   from fedot.api.main import Fedot
   from fedot.core.data.data import InputData
   from fedot.core.data.data_split import train_test_data_setup
   from fedot.core.pipelines.pipeline_builder import PipelineBuilder
   from fedot.core.repository.tasks import Task, TaskTypesEnum, TsForecastingParams
   ```

2. **Compose your custom pipeline**:
   ```python
   pipeline = PipelineBuilder() \
       .add_sequence('lagged', 'ridge', branch_idx=0) \
       .add_sequence('glm', branch_idx=1).join_branches('ridge').build()
   ```

3. **Specify the task and forecast length**:
   ```python
   task = Task(TaskTypesEnum.ts_forecasting,
               TsForecastingParams(forecast_length=10))
   ```

4. **Load data from a CSV file**:
   ```python
   train_input = InputData.from_csv_time_series(task=task,
                                                file_path='time_series.csv',
                                                delimiter=',',
                                                target_column='value')
   ```

5. **Split the data into train and test sets**:
   ```python
   train_data, test_data = train_test_data_setup(train_input)
   ```

6. **Initialize the model for time-series forecasting**:
   ```python
   model = Fedot(problem='ts_forecasting', task_params=task.task_params)
   ```

7. **Fit the pipeline**:
   ```python
   model.fit(train_data, predefined_model=pipeline)
   ```

8. **Obtain a one-step out-of-sample forecast**:
   ```python
   forecast = model.forecast(test_data)
   print(model.get_metrics(metric_names=['rmse', 'mae', 'mape'], target=test_data.target))
   ```

9. **Plot the forecasting result**:
   ```python
   model.plot_prediction()
   ```

This example demonstrates how to manually create and fit a custom pipeline for time-series forecasting using Fedot. The pipeline includes a `lagged` transformation followed by `ridge` regression in one branch, and a `glm` (generalized linear model) in another branch, with both branches' outputs combined using another `ridge` model for the final prediction.

**Comment by valer1435** [comment]
What context did you used to answer last question? @RepoPilotAssistant

**Comment by valer1435** [comment]
What context did you used to answer last question? @RepoPilotAssistant

**Comment by valer1435** [comment]
@RepoPilotAssistant thank you

**Comment by valer1435** [comment]
Which parameters Fedot.predict has?

**Comment by valer1435** [comment]
@RepoPilotAssistant Which parameters Fedot.predict has?

**Comment by valer1435** [comment]
@RepoPilotAssistant Which parameters Fedot.predict has?

**Comment by valer1435** [comment]
@RepoPilotAssistant Which parameters Fedot.predict has?

**Comment by valer1435** [comment]
@RepoPilotAssistant Which parameters Fedot.predict has?

**Comment by valer1435** [comment]
@RepoPilotAssistant Which parameters Fedot.predict has?

**Comment by valer1435** [comment]
@RepoPilotAssistant Which parameters Fedot.predict has?

**Comment by RepoPilotAssistant** [comment]
The `predict` method in Fedot has the following parameters:

- `features`: An array with features of the test data.
- `save_predictions`: A boolean indicating whether to save the predictions as a CSV file in the working directory.
- `in_sample`: A boolean used for time-series prediction. If `True`, it performs in-sample forecasting using the features with the number of iterations specified in `validation_blocks`.
- `validation_blocks`: The number of validation blocks for in-sample forecasting.

The method returns an array with the prediction values.

To go further, please follow links below:
1. [fedot\api\main.py](https://github.com/aimclub\FEDOT\blob/master\fedot\api\main.py)
2. [fedot\api\main.py](https://github.com/aimclub\FEDOT\blob/master\fedot\api\main.py)
3. [fedot\core\operations\evaluation\operation_implementations\models\ts_implementations\naive.py](https://github.com/aimclub\FEDOT\blob/master\fedot\core\operations\evaluation\operation_implementations\models\ts_implementations\naive.py)
4. [fedot\core\operations\evaluation\operation_implementations\models\ts_implementations\arima.py](https://github.com/aimclub\FEDOT\blob/master\fedot\core\operations\evaluation\operation_implementations\models\ts_implementations\arima.py)
5. [fedot\core\pipelines\prediction_intervals\utils.py](https://github.com/aimclub\FEDOT\blob/master\fedot\core\pipelines\prediction_intervals\utils.py)
